marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.mjs

```diff
@@ -106,6 +126,42 @@ export function setArtifact(artifactType, artifactName, artifact) {
     return artifact;
 }
 
+export function getArtifactSettings(artifactType, artifactName, artifactVersion = 'latest') {
+    const artifactSettingKey = generateArtifactKey(artifactType + 'Settings', artifactName);
+    if (!cachedArtifacts[artifactSettingKey]) {
+        const settingLibrary = getArtifactTypeLibrary(artifactType);
+        const settingNode = settingLibrary.getArtifactSettingNode(artifactName, artifactVersion);
+
+        if (fn.empty(settingNode)) {
+            return {};
+        }
+        cachedArtifacts[artifactSettingKey] = settingNode.toObject();
+    }
+    return cachedArtifacts[artifactSettingKey];
+}
+
+export function setArtifactSettings(artifactType, artifactName, settings) {
```

Comment:
Given that this is setting collections that are specific to loadData, shouldn't this have a loadData-specific name?

Reply:
you mean we can use 

http://marklogic.com/data-hub/load-data-artifact/settings/{loadArtifactName} insteads of http://marklogic.com/data-hub/load-data-artifact/settings ?

It should be better for query.

Reply:
@hao1st  It think what is meant is the collection more like `http://marklogic.com/data-hub/${artifactType}/settings`. Perhaps we could get the collection from the artifactLibrary and append "/settings" to them to stay consistent.

Reply:
@ryanjdew  make sense.